code/io/joy-sdl.cpp

```diff
@@ -351,7 +402,11 @@ namespace joystick
 			// Hat is inactive
 			return 0.0f;
 
-		} // Hat is active
+		} if (_hat[index].Value != hatIdxToEnum(pos)) {
```

Comment:
Should this be an `else if`? If it shouldn't then the `if` should not be on the same line as the closing brace.